From 38a33127f184e4426ed5fee9dfcf607e2beea355 Mon Sep 17 00:00:00 2001
Message-Id: <38a33127f184e4426ed5fee9dfcf607e2beea355.1413837857.git.chang-joon.lee@intel.com>
In-Reply-To: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
References: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
From: Deepak S <deepak.s@intel.com>
Date: Thu, 24 Jul 2014 15:17:05 +0530
Subject: [PATCH 248/312] FOR_UPSTREAM [VPG]: drm/i915: Add rc6 Debugfs
 support to CHV

This patch help to test rc6 enable/disable functionality at runtime.

Issue: GMIN-3326
Change-Id: I46dd3e5a48926a19d2906def3a10d2936e4b4747
Signed-off-by: Deepak S <deepak.s@intel.com>
---
 drivers/gpu/drm/i915/intel_pm.c |    8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_pm.c b/drivers/gpu/drm/i915/intel_pm.c
index e30f995..13f018b 100644
--- a/drivers/gpu/drm/i915/intel_pm.c
+++ b/drivers/gpu/drm/i915/intel_pm.c
@@ -3890,7 +3890,8 @@ static void cherryview_disable_rps(struct drm_device *dev)
 {
 	struct drm_i915_private *dev_priv = dev->dev_private;
 
-	I915_WRITE(GEN6_RC_CONTROL, 0);
+	/* Disable rc6 */
+	vlv_set_rc6_mode(dev, true);
 
 	gen8_disable_rps_interrupts(dev);
 }
@@ -4684,9 +4685,12 @@ static void cherryview_enable_rps(struct drm_device *dev)
 	DRM_DEBUG_DRIVER("PCBR offset : 0x%x\n", pcbr);
 
 	/* 3: Enable RC6 */
+	rc6_mode = GEN7_RC_CTL_TO_MODE;
+	dev_priv->rps.rc6_mask = rc6_mode;
+
 	if ((intel_enable_rc6(dev) & INTEL_RC6_ENABLE) &&
 						(pcbr >> VLV_PCBR_ADDR_SHIFT))
-		rc6_mode = GEN7_RC_CTL_TO_MODE;
+		vlv_set_rc6_mode(dev, false);
 
 	I915_WRITE(GEN6_RC_CONTROL, rc6_mode);
 
-- 
1.7.9.5

